book/crud.rst

```diff
@@ -0,0 +1,477 @@
+    ``acl_resource`` option can be used to define access rules for a datagrid. In this example, the
+    ACL is not needed as the controller itself is already protected with the same rule. Though it
+    is recommended to add it nonetheless since a datagrid can be reused in other actions.
+
+``columns``
+    The ``columns`` key is used to configure the grid's columns. The ``label`` option is used to
+    define a headline per column (the Twig ``trans`` filter is automatically applied to each
+    label). If you do not configure a ``frontend_type``, the value will be shown as is.
+
+``properties``
+    With ``properties`` you have to configure some reusable properties. At least, you will have to
+    define the ``id`` property which is used by the grid to determine if an entity is new. Usually,
+    you can set this to ``~``. It will then be assumed that your entity is identified by an ``id``
+    property. Usually, you also configure the links to other controller actions.
+
+``sorters``
+    To be able to click on the column headlines to sort the rows, you use the ``sorters`` option.
+    For each column you define the entity property to sort by when the column is clicked. The
+    ``default`` key is used to determine how to sort rows when the datagrid is visited the first
+    time.
+
+``filters``
+    Configures the filters displayed above the datagrid.
+
+``actions``
+    Here, you define which actions can be performed by the user. For the links, you refer to the
+    ones defined with the ``properties`` key before. Labels will also be translated. If the
+    configured ``type`` is ``navigate``, clicking the icon will be the same as clicking on an HTML
+    link while using ``delete`` performs an HTTP DELETE request in the background.
+
+.. _book-crud-create-update:
+
+Creating and Updating Entities
+------------------------------
+
+To be able to create new vehicles and update existing ones, you first have to create a form type
+and register it as a service:
+
+.. code-block:: php
+    :linenos:
+
+    // src/InventoryBundle/Form/Type/VehicleType.php
+    namespace InventoryBundle\Form\Type;
+
+    use Symfony\Component\Form\AbstractType;
+    use Symfony\Component\Form\FormBuilderInterface;
+    use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+    class VehicleType extends AbstractType
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('model')
+                ->add('seats')
+                ->add('boughtAt')
+                ->add('leasedUntil')
+            ;
+        }
+
+        public function setDefaultOptions(OptionsResolverInterface $resolver)
+        {
+            $resolver->setDefaults(array(
+                'data_class' => 'InventoryBundle\Entity\Vehicle',
+            ));
+        }
+
+        public function getName()
+        {
+            return 'inventory.vehicle';
```

Comment:
Form type name ("inventory.vehicle") is not valid. Names must only contain letters, numbers, and "_".